syft/execution/protocol.py

```diff
@@ -1,479 +1,560 @@
+                result_role.register_output(result)
+
+        self.is_built = True
+
+        # Build registered translations
+        # for translator in Protocol._build_translators:
+        #     try:
+        #         self.add_translation(translator)
+        #         self.translations.append(translator)
+        #     except:
+        #         warnings.warn(f"Failed to translate Protocol with {translator}")
+
+        return results
+
+    def toggle_tracing(self, value=None):
+        self.tracing = value if value is not None else not self.tracing
+        # self.state.tracing = self.tracing
+        for role in self.roles.values():
+            for ph in role.placeholders.values():
+                ph.tracing = self.tracing
+
+    def copy(self):
+        """Creates a copy of a protocol."""
+        protocol_copy = Protocol(
+            name=self.name,
+            roles={role_id: role.copy() for role_id, role in self.roles.items()},
+            include_state=self.include_state,
+            is_built=self.is_built,
+            id=sy.ID_PROVIDER.pop(),
+            owner=self.owner,
+            tags=self.tags,
+            description=self.description,
+        )
 
-    def run(self, *args, **kwargs):
+        protocol_copy.torchscript = self.torchscript
+
+        return protocol_copy
+
+    # def __setattr__(self, name, value):
+    #     """Add new tensors or parameter attributes to the state and register them
+    #     in the owner's registry
+    #     """
+    #     if isinstance(value, torch.jit.ScriptModule):
+    #         object.__setattr__(self, name, value)
+    #     elif isinstance(value, FrameworkTensor):
+    #         self.role.register_state_tensor(value, self.owner)
+    #         self.state_attributes[name] = value
+    #     elif isinstance(value, FrameworkLayerModule):
+    #         for param in value.parameters():
+    #             self.role.register_state_tensor(param, self.owner)
+    #         self.state_attributes[name] = value
+    #     else:
+    #         object.__setattr__(self, name, value)
+
+    # def __getattr__(self, name):
+    #     if name not in self.state_attributes:
+    #         raise AttributeError("State attribute not found.")
+
+    #     value = self.state_attributes[name]
+    #     if not self.is_building:
+    #         return value
+
+    #     if isinstance(value, FrameworkTensor):
+    #         return self.role.placeholders[value.id]
+    #     elif isinstance(value, FrameworkLayerModule):
+    #         # We need to deepcopy here otherwise the real layer is modified when the Protocol is being built
+    #         copied_layer = copy.deepcopy(value)
+    #         for copied_param, param in zip(copied_layer.named_parameters(), value.parameters()):
+    #             (copied_name, _) = copied_param
+    #             copied_layer._parameters[copied_name] = self.role.placeholders[param.id]
+
+    #         return copied_layer
```

Comment:
I have never entirely understood what these methods are doing in `Plan`, but they seem like they exist for convenience? If so, let's remove them here and worry about that later.
```suggestion
```

Reply:
They're here for the class syntax. You can build a plan that looks like a torch Network with them (writing stuff as `self.fc1 = something` in `__init__`)